jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.cpp

```diff
@@ -733,25 +734,58 @@ ecma_date_make_day (ecma_number_t year, /**< year value */
     return ecma_number_make_nan ();
   }
 
+  /* 2., 3., 4. */
   ecma_number_t y = ecma_number_trunc (year);
   ecma_number_t m = ecma_number_trunc (month);
   ecma_number_t dt = ecma_number_trunc (date);
+  /* 5. */
   ecma_number_t ym = y + (ecma_number_t) floor (m / 12);
+  /* 6. */
   ecma_number_t mn = (ecma_number_t) fmod (m, 12);
   mn = (mn < 0) ? 12 + mn : mn;
-  ecma_number_t time = ecma_date_time_from_year (ym);
 
-  JERRY_ASSERT (ecma_date_year_from_time (time) == ym);
+  /* 7. We need to find the time where: YearFromTime(time) == ym
+   *    and MonthFromTime(time) == mn and DateFromTime(time) == 1.
+   *    If this is not possible return NaN.
+   */
```

Comment:
@kkristof, the comment is requirement for algorithm below (I think description of the requirement could be replaced with `/* 7. */` comment in the case).

However, there is no explanation of why the algorithm below does really find value, described in the comment / specification, i.e. there is no description of why the steps performed below are equivalent to behaviour, specified in 15.9.1.12 step 7.